test/Groups/fpgroup.jl

```diff
@@ -0,0 +1,55 @@
+@testset "fp group elements letters" begin
+  F = @free_group(:F1, :F2)
+  x = F1^5 * F2^-3
+  l_1 = [1, 1, 1, 1, 1, -2, -2, -2]
+  
+  # test trivial cases
+  @test letters(one(F)) == []
+  @test F(Vector{Int}()) == one(F)
+  
+  # test general letters usage
+  @test letters(x) == l_1
+  @test F(l_1) == x
+
+  # test for generator number check
+  l_2 = [l_1 ; [3, 4]]
+  @test_throws ArgumentError F(l_2)
+
+  G, epi = quo(F, [F1^10, F2^10])
+  
+  # test trivial cases for quotient
+  @test letters(one(G)) == []
+  @test G(Vector{Int}()) == one(G)
+
+  # test general letters usage for quotient
+  @test letters(epi(x)) == l_1
```

Comment:
this must not hold, as the group could choose to use a different internal representation of `epi(x)`.
I think the only invariant we can check here is
```suggestion
  @test G(letters(epi(x))) == G(l_1)
```